robyn/__init__.py

```diff
@@ -31,19 +27,15 @@ def __init__(self, file_object: str) -> None:
         directory_path = os.path.dirname(os.path.abspath(file_object))
         self.file_path = file_object
         self.directory_path = directory_path
-        self.parser = ArgumentParser()
-        self.dev = self.parser.is_dev
-        self.processes = self.parser.num_processes
-        self.workers = self.parser.workers
-        self.log_level = self.parser.log_level
+        self.config = Config()
```

Comment:
speaking of testing, a step in the direction of improving the integration testing ability, we can start improving the inversion of control by changing the constructor to something like this

```
def __init__(self, file_object:str, router: Router=Router(), middleware_router:MiddlewareRouter(), ..):
           self.router = router
           self.config=config
           
           and so on..
```

we can address this issue in this PR or the next one. However, makes more sense to you

Reply:
I think we can add this to a future PR if you want, I am not sure what kind of tests you want to write with this

Reply:
Yep, sounds good.